CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/gui/VisualGraphTopComponent.java

```diff
@@ -956,20 +949,27 @@ protected void handleSave() throws IOException {
                 // graph will get saved each time.
                 requestActive();
 
-                SaveAsAction action = new SaveAsAction();               
+                final SaveAsAction action = new SaveAsAction();
                 action.actionPerformed(null);
                 isSaved = action.isSaved();
                 return;
             }
 
-            final String name = gdo.getName();
-
-            // Create a new file and write to it.
-            final String tmpnam = String.format("%s_tmp%08x", name, gdo.hashCode());
-            final GraphDataObject freshGdo = (GraphDataObject) gdo.createFromTemplate(gdo.getFolder(), tmpnam);
-            final BackgroundWriter writer = new BackgroundWriter(name, freshGdo, true);
-            writer.execute();
-            isSaved = true;
+            if (gdo.isValid()) {
+                final String name = gdo.getName();
+                // Create a new file and write to it.
+                final String tmpnam = String.format("%s_tmp%08x", name, gdo.hashCode());
+                final GraphDataObject freshGdo = (GraphDataObject) gdo.createFromTemplate(gdo.getFolder(), tmpnam);
+                final BackgroundWriter writer = new BackgroundWriter(name, freshGdo, true);
+                writer.execute();
+                isSaved = true;
+            } else {
+                // File might have been saved over, so just open up the 'save as' menu
+                requestActive();
+                final SaveAsAction action = new SaveAsAction();
+                action.actionPerformed(null);
+                isSaved = action.isSaved();
+            }
```

Comment:
does this check still need to be in place with the updates you made? Looking good in testing

Reply:
The file being overwritten shouldn't happen anymore, but there might be other cases where a graph is "invalid". Up to you if you think it should go.

Reply:
There is one case I've discovered where the code does come into effect (which I will put into another ticket) but it only kicks in if you make changes afterwards. Can probably leave in for now but we should probably make a note that when all cases of outside manipulation are covered, we probably won't have any reason to retain the code